sonarqube-companion-frontend/src/app/config/integrations/ldap/sq-settings-ldap-integrations.component.ts

```diff
@@ -0,0 +1,157 @@
+                </div>
+                <div class="item">
+                  <label>Id attribute</label>
+                  <input type="text" [(ngModel)]="ldapConfig.members.idAttribute">
+                </div>
+                <div class="item">
+                  <label>First name attribute</label>
+                  <input type="text" [(ngModel)]="ldapConfig.members.firstNameAttribute">
+                </div>
+                <div class="item">
+                  <label>Last name attribute</label>
+                  <input type="text" [(ngModel)]="ldapConfig.members.lastNameAttribute">
+                </div>
+                <div class="item">
+                  <label>Mail attribute</label>
+                  <input type="text" [(ngModel)]="ldapConfig.members.mailAttribute">
+                </div>
+                <div>
+                  <sq-settings-badge [(items)]="ldapConfig.members.aliasesAttributes"
+                                     [title]="'Aliases attributes'"></sq-settings-badge>
+                </div>
+              </div>
+            </div>
+
+
+            <div class="section">
+              <label>Groups</label>
+              <div>
+                <div class="item">
+                  <label>Group object class</label>
+                  <input type="text" [(ngModel)]="ldapConfig.groups.groupObjectClass">
+                </div>
+                <div class="item">
+                  <label>Group name attribute</label>
+                  <input type="text" [(ngModel)]="ldapConfig.groups.groupNameAttribute">
+                </div>
+                <div class="item">
+                  <label>Membership attribute</label>
+                  <input type="text" [(ngModel)]="ldapConfig.groups.membershipAttribute">
+                </div>
+                <div class="groups">
+                  <sq-settings-map
+                    [title]="'Group mappings'"
+                    [loaded]="loaded"
+                    [(data)]="ldapConfig.groups.groupMappings"
+                  ></sq-settings-map>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </ng-container>
+    </div>
+  `
+})
+
+export class SettingsLdapIntegrationsComponent implements AfterViewInit {
+
+  ldapConfig: LdapConfig;
+  loaded: boolean = false;
+  groups: GroupLightModel[] = [];
+  errorMsg: string = '';
+
+  constructor(private ldapIntegrationService: LdapIntegrationService) {
+  }
+
+  ngAfterViewInit(): void {
+    this.load();
+  }
+
+  save(): void {
+    this.ldapIntegrationService.updateLdapConfig(this.ldapConfig).subscribe(data => {
+      if (data.valid) {
+        this.load();
+        this.errorMsg = '';
+      } else {
+        this.errorMsg = data.message;
+      }
+    });
+  }
+
+  private load(): void {
+    this.loaded = false;
+    this.ldapIntegrationService.getLdapConfig().subscribe(data => {
```

Comment:
async do widoku i loaded może wynikać z ngIf z else

Reply:
zwłaszcza że nie pamiętasz o unsubscribe żadnym, nigdzie

Reply:
tu jeszcze kminię, bo nie do końca może to wynikać z ngif, bo po zapisie chcę pokazać loader znowu, więc gdzieś stan muszę trzymać

Reply:
done